Source/WebCore/accessibility/AXTextMarker.cpp

```diff
@@ -732,20 +740,40 @@ const AXTextRuns* AXTextMarker::runs() const
     return object ? object->textRuns() : nullptr;
 }
 
-AXTextMarker AXTextMarker::findMarker(AXDirection direction, std::optional<AXID> stopAtID) const
+AXTextMarker AXTextMarker::findMarker(AXDirection direction, bool coalesceObjectBreaks, bool ignoreBRs, std::optional<AXID> stopAtID) const
 {
+    // This method has two boolean options:
+    // - coalesceObjectBreaks: Mimics behavior from textMarkerDataForNextCharacterOffset, where we skip nodes
+    //   that have the same visual position (i.e., there is 0 length between them). When false, we traverse all
+    //   possible text markers (which is important for searching)
+    // - ignoreBRs: In most cases, we want to skip <br> tags when not in an editable context. This is not true,
+    //   for example, when computing text marker indexes.
+
     if (!isValid())
         return { };
     if (!isInTextRun())
-        return toTextRunMarker().findMarker(direction, stopAtID);
+        return toTextRunMarker().findMarker(direction, coalesceObjectBreaks, ignoreBRs, stopAtID);
 
-    if ((direction == AXDirection::Next && offset() < runs()->totalLength()) || (direction == AXDirection::Previous && offset() > 0))
-        return { treeID(), objectID(), direction == AXDirection::Next ? offset() + 1 : offset() - 1 };
+    RefPtr object = isolatedObject();
+
+    // If the BR isn't in an editable ancestor, we shouldn't be including it (in most cases of findMarker).
+    bool shouldSkipBR = (ignoreBRs && !object->editableAncestor()) && object && object->roleValue() == AccessibilityRole::LineBreak;
+    if (!shouldSkipBR
+        && ((direction == AXDirection::Next && offset() < runs()->totalLength())
+        || (direction == AXDirection::Previous && offset()))) {
```

Comment:
Could we add a comment above this if-statement stating what it's checking for? It's got a lot of conditions, which is fine — just would be nice to have plain language describing what it's doing, if possible. Or we could create named `bool` variables for all / part of these conditions.

Reply:
Yeah, there are probably a few of these conditions that I can explain in a comment. I'll try to break it up + add context.